lib/tape.js

```
@@ -0,0 +1,259 @@
+ * @return string with test function name if transformations were made
+ */
+function updateTapeRequireAndImport(j, ast) {
+	let testFunctionName = null;
+	ast.find(j.CallExpression, {
+		callee: {name: 'require'},
+		arguments: arg => arg[0].value === 'tape'
+	})
+	.filter(p => p.value.arguments.length === 1)
+	.forEach(p => {
+		p.node.arguments[0].value = 'ava';
+		testFunctionName = p.parentPath.value.id.name;
+	});
+
+	ast.find(j.ImportDeclaration, {
+		source: {
+			type: 'Literal',
+			value: 'tape'
+		}
+	})
+	.forEach(p => {
+		p.node.source.value = 'ava';
+		testFunctionName = p.value.specifiers[0].local.name;
+	});
+
+	return testFunctionName;
+}
+
+export default function tapeToAva(fileInfo, api) {
+	const j = api.jscodeshift;
+	const ast = j(fileInfo.source);
+
+	const testFunctionName = updateTapeRequireAndImport(j, ast);
+
+	if (testFunctionName) {
+		const transforms = [
+			function detectUnsupportedNaming() {
+				// Currently we only support "t" as the test argument name
+				const validateTestArgument = p => {
+					const lastArg = p.value.arguments[p.value.arguments.length - 1];
+					if (lastArg && lastArg.params && lastArg.params[0]) {
+						const lastArgName = lastArg.params[0].name;
+						if (lastArgName !== 't') {
+							throw new Error(`Argument to test function should be named "t" not "${lastArgName}"`);
+						}
+					}
+				};
+
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: testFunctionName}
+					}
+				})
+				.forEach(validateTestArgument);
+
+				ast.find(j.CallExpression, {
+					callee: {name: testFunctionName}
+				})
+				.forEach(validateTestArgument);
+			},
+
+			function detectUnsupportedFeatures() {
+				ast.find(j.CallExpression, {
+					callee: {
+						object: {name: 't'},
+						property: ({name}) => unsupportedTestFunctions.has(name)
+					}
+				})
+				.forEach(p => {
+					throw new Error(`Codemod transformation of "${p.value.callee.property.name}" is not supported`);
```

Do we want to fail the transformation when unsupported features are found? I think it might be more helpful to log out `file.path` and the line/column info, and `console.warn` so the user knows to go make an update.